pkg/monitortests/clusterversionoperator/legacycvomonitortests/operators.go

```diff
@@ -606,6 +609,191 @@ func testOperatorStateTransitions(events monitorapi.Intervals, conditionTypes []
 	return ret
 }
 
+func clusterOperatorIsNotProgressingWhenMachineConfigIs(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	var ret []*junitapi.JUnitTestCase
+	upgradeWindows := getUpgradeWindows(events)
+
+	var machineConfigProgressing time.Time
+	var eventsInUpgradeWindows monitorapi.Intervals
+
+	var start, stop time.Time
+	for _, event := range events {
+		if !isInUpgradeWindow(upgradeWindows, event) {
+			continue
+		}
+		eventsInUpgradeWindows = append(eventsInUpgradeWindows, event)
+		if start.IsZero() || event.From.Before(start) {
+			start = event.From
+		}
+		if stop.IsZero() || event.To.After(stop) {
+			stop = event.To
+		}
+	}
+	duration := stop.Sub(start).Seconds()
+
+	eventsByOperator := getEventsByOperator(eventsInUpgradeWindows)
+	for _, mcEvent := range eventsByOperator["machine-config"] {
+		condition := monitorapi.GetOperatorConditionStatus(mcEvent)
+		if condition == nil {
+			continue // ignore non-condition intervals
+		}
+		if condition.Type == configv1.OperatorProgressing && condition.Status == configv1.ConditionTrue {
+			machineConfigProgressing = mcEvent.To
+			break
+		}
+	}
+
+	mcTestCase := &junitapi.JUnitTestCase{
+		Name:     fmt.Sprintf("[bz-Machine Config Operator] clusteroperator/machine-config must go Progressing=True during an upgrade test"),
+		Duration: duration,
+	}
+	if machineConfigProgressing.IsZero() {
+		mcTestCase.FailureOutput = &junitapi.FailureOutput{
+			Output: fmt.Sprintf("machine-config was never Progressing=True during the upgrade window from %s to %s", start.Format(time.RFC3339), stop.Format(time.RFC3339)),
+		}
+		return []*junitapi.JUnitTestCase{mcTestCase}
+	} else {
+		mcTestCase.SystemOut = fmt.Sprintf("machine-config became Progressing=True at %s during the upgrade window from %s to %s", machineConfigProgressing.Format(time.RFC3339), start.Format(time.RFC3339), stop.Format(time.RFC3339))
+	}
+	ret = append(ret, mcTestCase)
+
+	for _, operatorName := range platformidentification.KnownOperators.Difference(sets.NewString("machine-config")).List() {
+		bzComponent := platformidentification.GetBugzillaComponentForOperator(operatorName)
+		if bzComponent == "Unknown" {
+			bzComponent = operatorName
+		}
+		testName := fmt.Sprintf("[bz-%v] clusteroperator/%v should stay Progressing=False while MCO is Progressing=True", bzComponent, operatorName)
```

Comment:
The purpose of the [bz-COMPONENT] label is to clearly find the responsible bugzilla component, which may be internally mapped to a Jira compoent. If the component is unknown I wouldn't paste the operator name, as it does not fulfill this purpose (directly and confidenetly link a test case to a component). We doe paste the operatorName in the test name so that at least helps finding the respective compoenent.